tests/ui/lint/noop-method-call.stderr

```diff
@@ -105,5 +105,57 @@ LL + #[derive(Clone)]
 LL | struct PlainType<T>(T);
    |
 
-warning: 7 warnings emitted
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:70:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^ help: remove this redundant call
+   |
+   = note: the type `non_clone_types::NotClone` does not implement `Clone`, so calling `clone` on `&non_clone_types::NotClone` copies the reference, which does not do anything and can be removed
+
+warning: call to `.clone()` on a reference in this situation does nothing
+  --> $DIR/noop-method-call.rs:75:6
+   |
+LL |     v.clone();
+   |      ^^^^^^^^
+   |
+   = note: the type `non_clone_types::ConditionalClone<PlainType<u32>>` does not implement `Clone`, so calling `clone` on `&non_clone_types::ConditionalClone<PlainType<u32>>` copies the reference, which does not do anything and can be removed
+help: remove this redundant call
+   |
+LL -     v.clone();
+LL +     v;
+   |
+help: if you meant to clone `non_clone_types::ConditionalClone<PlainType<u32>>`, implement `Clone` for it
```

Comment:
The wording here is no longer quite accurate as we do not implement `Clone` for `ConditionalClone<PlainType<u32>>` anymore, but for `PlainType`. Not sure how to fix this.

Reply:
On `159` you are passing `orig_ty` without modification.

I think if you need to store the type name in [`144`](https://github.com/rust-lang/rust/pull/134903/files/743799880a1c1091dd5cd32e7044bc90c58838f5#diff-efe45c248e022b7a8ad064214427dad6129407be362155867ef3411ce31a1486R144-R146), and modify the error to have both the original type and the type that didn't impl `Clone` so the wording of the note also still makes sense ("the type `{$non_clone_ty}` does not implement `{$trait_}`, so calling `{$method}` on `&{$orig_ty}` copies the reference, which does not do anything and can be removed").